### PR TITLE
Add `--no-upgrade` option.

### LINKED ISSUE #229: Allow only installing missing formulae without upgrading existing ones
I'm a big fan of having a Brewfile in my home directory (and keeping it in version control) and using `brew bundle --global` to ensure I have all the utilities and apps I need. However, because my Brewfile lists formulae I use for development such as `node` and `go`, I don't want these to be auto-upgraded when I run `brew bundle` to fill in the blanks, i.e. install the new additions I made to my Brewfile. I only ever want formulae to be upgraded when I issue an explicit `brew upgrade`.

Now, I know about pinning and yes, I could easily `brew pin node go` to prevent them from being auto-upgraded. But, on principle, I kind of don't want `brew bundle --global` to ever upgrade *any* of the formulae. That means I would have to pin them all, which would make manual upgrades more clumsy since I'd have to go through an unpin-upgrade-repin dance.

I also know about `homebrew/versions`, and while I see it very useful for project-specific Brewfiles shared between a team of people, I feel it's an overkill for my own personal needs, where I simply want to install the latest version of e.g. `node` when I set up a fresh computer, but have control over when it's upgraded.

I've created [a workaround for myself](https://github.com/mislav/dotfiles/commit/7a05ec98779b25981c91526f19782aaaf274c183) that temporarily pins everything before `brew bundle`, and restores previously pinned on completion, but that makes me wish that `brew bundle` had some kind of `--no-upgrade` option.

Does this feature make sense for anyone else, or was `brew bundle` envisioned with a different set of principles in mind?

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
As requested for people who want to avoid upgrading.

CC @mislav @jeremy

Fixes #229.

### CODE REVIEW

File: lib/bundle/brew_installer.rb [hunk 1]
```diff
@@ -61,7 +62,9 @@ def service_change_state!
     end
 
     def self.formula_installed_and_up_to_date?(formula)
-      formula_installed?(formula) && !formula_upgradable?(formula)
+      return false unless formula_installed?(formula)
+      return true if ARGV.include?("--no-upgrade")
```
Comment (mislav) [inline lib/bundle/brew_installer.rb]:
```rb
formula_installed?(formula) && (
  ARGV.include?("--no-upgrade") || !formula_upgradable?(formula)
)
```

Comment (MikeMcQuaid) [inline lib/bundle/brew_installer.rb]: I prefer the multiple return approach; I think it's easier to parse quickly.

**Comment by javierjulio** [comment]
@MikeMcQuaid this is great, thanks! Just like @mislav I needed the same thing as most dependencies I have are important and I'd like to upgrade explicitly.

It is a little alarming now when using the `--no-upgrade` flag and seeing "Succeeded in installing..." in the output for each brew call when tap, cask and mas show "Skipping install of..." when the `-v` is given. Would you accept a PR to include this for brew installer?

**Comment by MikeMcQuaid** [comment]
@javierjulio We based the output here on that of Bundler which behaves similarly. I'm not entirely sure what you're asking?

**Comment by mislav** [comment]
@MikeMcQuaid Bundler says `Using foo 1.0.0` when not installing/upgrading, and `Installing foo …` when it's actually installing. I think there is value in output accurately describing what's being done.

**Comment by MikeMcQuaid** [comment]
TIL, thanks @mislav. @javierjulio Yes, we'd accept a PR to say "Using" instead of "Installing" in that case.

**Pull request merged**